pallets/claims/src/lib.rs

```diff
@@ -167,3 +164,44 @@ fn to_ascii_hex(data: &[u8]) -> Vec<u8> {
 	}
 	r
 }
+
+/// Signed extension that checks for the `claim` call and in that case, it verifies an Ethereum signature
+#[derive(Encode, Decode, Clone, Eq, PartialEq)]
+pub struct ValidateClaim<T: Config + Send + Sync>(PhantomData<T>);
+
+impl<T: Config + Send + Sync> sp_std::fmt::Debug for ValidateClaim<T> {
+	fn fmt(&self, f: &mut sp_std::fmt::Formatter) -> sp_std::fmt::Result {
+		write!(f, "ValidateClaim")
+	}
+}
+
+impl<T: Config + Send + Sync> SignedExtension for ValidateClaim<T>
```

Comment:
Can we write test for this?

Reply:
For what exactly you mean? 

There are tests of signed extension - all 3 cases covered.

Reply:
If the tx gets thrown out at the right moment.